packages/did-resolver/src/base-resolver.ts

```diff
@@ -36,16 +36,19 @@ export abstract class BaseResolver {
   ): Promise<DidDocument | null> {
     if (this.cache && !forceRefresh) {
       const fromCache = await this.cache.checkCache(did)
-      if (fromCache?.expired) {
-        this.refreshCache(did)
+      if (fromCache?.stale) {
+        await this.refreshCache(did)
```

Comment:
I don't think we necessarily need the `await` here since the stale value is good to be served immediately while it gets refreshed.

Reply:
I went back & forth on this. but i decided to keep the await there & then the cache gets to decide how async it should be

This simple memory cache does not do it async. but the sql cache that we use on the appview keeps a pqueue that it pushes it on & does the refresh async

Reply:
the other option is to get the pqueue into the base-resolver but that all seemed a little bit heavy

Reply:
Seems good as-is 👍